app/views/admin/index.scala.html

```diff
@@ -661,6 +664,9 @@ <h1>Labels</h1>
                                         <td>@label.temporary</td>
                                         <td>@label.tags.mkString(", ")</td>
                                         <td>@label.description</td>
+                                        <td>@LabelTable.getLabelMetadata(label.labelId).validations("agree")</td>
```

Comment:
I think this will actually make the admin page load much slower on large databases. This adds 3 queries for every label in the table. There are 5000 labels, so that's an extra 15,000 queries! I tested on my local environment and the admin page went from loading in 55 seconds on develop to 7 minutes and 30 seconds with these changes :sweat_smile:  It would be better to modify the main query in the `selectTopLabelsAndMetadata` function to include the validations. Making that change to the query is pretty complicated so I worked on it a bit myself and have some stuff that should help out.

Here are the changes I think you'll want to make. You'll add a subquery to get the number of validations:
```
(
    SELECT label.label_id, validation_option_id, COUNT(label_validation_id) AS count
    FROM label
    FULL JOIN validation_options ON TRUE
    LEFT JOIN label_validation ON label.label_id = label_validation.label_id
        AND label_validation.validation_result = validation_options.validation_option_id
    WHERE label.deleted = FALSE
    GROUP BY label.label_id, validation_option_id
) as val
```

Join using this line: `AND lb1.label_id = val.label_id`

You'll need to include the new relevant columns in the SELECT statement:
```
val.validation_option_id,
val.count
```

So the full query will looks like this:
```
    val selectQuery = Q.query[Int, (Int, String, Boolean, String, Float, Float, Int, Int, Int, Int, Int,
      Int, String, String, Option[java.sql.Timestamp], String, String, Option[Int], Boolean,
      Option[String], Int, Int)](
      """SELECT lb1.label_id,
        |       lb1.gsv_panorama_id,
        |       lb1.tutorial,
        |       gsv_data.image_date,
        |       lp.heading,
        |       lp.pitch,
        |       lp.zoom,
        |       lp.canvas_x,
        |       lp.canvas_y,
        |       lp.canvas_width,
        |       lp.canvas_height,
        |       lb1.audit_task_id,
        |       u.user_id,
        |       u.username,
        |       lb1.time_created,
        |       lb_big.label_type,
        |       lb_big.label_type_desc,
        |       lb_big.severity,
        |       lb_big.temp,
        |       lb_big.description,
        |       val.validation_option_id,
        |       val.count
        |FROM sidewalk.label AS lb1,
        |     sidewalk.gsv_data,
        |     sidewalk.audit_task AS at,
        |     sidewalk_user AS u,
        |     sidewalk.label_point AS lp,
        |     (
        |         SELECT lb.label_id,
        |                lb.gsv_panorama_id,
        |                lbt.label_type,
        |                lbt.description AS label_type_desc,
        |                sev.severity,
        |                COALESCE(lab_temp.temporary, 'FALSE') AS temp,
        |                lab_desc.description
        |					FROM label AS lb
        |				  LEFT JOIN sidewalk.label_type as lbt ON lb.label_type_id = lbt.label_type_id
        |  				LEFT JOIN sidewalk.label_severity as sev ON lb.label_id = sev.label_id
        |				  LEFT JOIN sidewalk.label_description as lab_desc ON lb.label_id = lab_desc.label_id
        |				  LEFT JOIN sidewalk.label_temporariness as lab_temp ON lb.label_id = lab_temp.label_id
        |     ) AS lb_big,
        |     (
        |        SELECT label.label_id, validation_option_id, COUNT(label_validation_id) AS count
        |        FROM label
        |        FULL JOIN validation_options ON TRUE
        |        LEFT JOIN label_validation ON label.label_id = label_validation.label_id
        |            AND label_validation.validation_result = validation_options.validation_option_id
        |        WHERE label.deleted = FALSE
        |        GROUP BY label.label_id, validation_option_id
        | ) AS val
        | WHERE lb1.deleted = FALSE
        |    AND lb1.gsv_panorama_id = gsv_data.gsv_panorama_id
        |    AND lb1.audit_task_id = at.audit_task_id
        |    AND lb1.label_id = lb_big.label_id
        |    AND at.user_id = u.user_id
        |    AND lb1.label_id = lp.label_id
        |    AND lb1.label_id = val.label_id
        | ORDER BY lb1.label_id DESC
        | LIMIT ?""".stripMargin
    )
```

Then there's a few other things you'll need to do. We can't make a column for each validation result easily using SQL, so we actually will have 15,000 rows instead of 5,000 rows as a result of this query (so make sure to query for n * 5 instead of just n). So each label has one row for agree, one for disagree, and one for unsure. Now what we want to do is turn that into one row for each label, with a map from the validation result to the number of those validations (same as the output of `getLabelMetadata(label.labelId).validations`.

To do this, we'll group by the label id, create a map from those last two columns we just added, and carry along all the rest of the columns (which should are identical for the three rows for the same label_id). Here is an example of what I mean in a more simplified way... Assume the tuples below have the following values (labelId, auditTaskId, validationOptionId, count). The auditTaskId is a stand in for all the columns that will need to be carried over because they are the same for the same label_id.
```
val lst = List((1, 10, 1, 4), (1, 10, 2, 0), (1, 10, 3, 2), (2, 15, 1, 3), (2, 15, 2, 5), (2, 15, 3, 0))
println(lst.groupBy(_._1).map{ case (y, group) => (y, group.head._2, group.map(z => (z._3, z._4)).toMap) }.toList)
// Output: List((2,15,Map(1 -> 3, 2 -> 5, 3 -> 0)), (1,10,Map(1 -> 4, 2 -> 0, 3 -> 2)))
```

Then there's a few other things you'll need to do as you go along. Like modifying or making a new function like `labelAndValidationsToMetadata` to use the right set of parameters.